### ISSUE TITLE
Don't highlight in insert mode over word

### ISSUE THREAD

**Issue by seblj** [lewis6991/spellsitter.nvim#59]
So I don't know if this is really possible to achieve, but I wanted to open an issue about it to hear if it is possible to do.

With the built-in spellchecker, it is not highlighted while in insertmode and typing out the word. See video below

https://user-images.githubusercontent.com/5160701/152340714-63fed0c5-5e7e-42e9-b95b-de785ee3f89a.mov

Here is the same thing with `spellsitter.nvim` enabled:

https://user-images.githubusercontent.com/5160701/152340787-2d7a394f-8a20-4913-8faf-2e922e2d3f70.mov

Is it possible to make it not highlight while the cursor is over the word and in insert mode? It's a bit distracting to see it highlight the word as bad spelling while typing out the word

**Comment by seblj** [comment]
So I found out it is possible by looking at the code. By doing this, it works as expected for me:

```Lua
local function add_extmark(bufnr, lnum, col, len, highlight)
  local cursor = vim.api.nvim_win_get_cursor(0)
  if vim.fn.mode() == 'i' and cursor[1]-1 == lnum and col <= cursor[2] and col+len >= cursor[2] then
    return
  end

  ...
end
```

I don't know if you would like this to be the default behaviour to make it more similar to how the built-in spellchecker works, but it would be really nice if it could be an option. Let me know what you think! I can also send a PR with this, and also make it an option if you don't want to change the default behaviour!

**Comment by lewis6991** [comment]
The goal is to mimic the built-in spellchecker as much as possible so go ahead and open a PR.

**Comment by seblj** [comment]
Okay will do! BTW, not related to this. I noticed, that spellsitter doesn't capture the SpellCap like the built-in spellchecker. I am currently trying to investigate why it doesn't do that. Right now it looks like `spellbadword` doesn't return it, and I'll try to see if I can find out why

**Comment by seblj** [comment]
Yes

**Comment by lewis6991** [comment]
Won't be related to `spellbadword()` then.

**Comment by seblj** [comment]
> Won't be related to `spellbadword()` then.

Okay, doesn't work with `spellbadword()`or `vim.spell()` then because I tried both

With spellsitter and highlighting with treesitter:
<img width="257" alt="image" src="https://user-images.githubusercontent.com/5160701/152352824-1ea56b46-4275-4ca4-871c-64440b02d8c2.png">

After running TSBufDisable highlight

<img width="236" alt="image" src="https://user-images.githubusercontent.com/5160701/152352990-be758080-1976-41c2-ab53-62d83161596a.png">

**Comment by lewis6991** [comment]
Yeah, `SpellCap` is only applied in `vim.spell` if it appears after `. `. `syntax.c/spell.c/screen.c` probably have some special logic to check in more places.
